pkg/tests/api/alerting/api_ruler_test.go

```diff
@@ -425,3 +425,348 @@ func TestAlertRuleConflictingTitle(t *testing.T) {
+
+		body, _ := rulesNamespaceWithoutVariableValues(t, b)
+		require.JSONEq(t, expectedFilteredByJSON, body)
+	}
+
+	// Now, let's check we get no rules when filtering by an unknown dashboard_uid
+	{
+		promRulesURL := fmt.Sprintf("http://grafana:password@%s/api/ruler/grafana/api/v1/rules?dashboard_uid=%s", grafanaListedAddr, "abc")
+		// nolint:gosec
+		resp, err := http.Get(promRulesURL)
+		require.NoError(t, err)
+		t.Cleanup(func() {
+			err := resp.Body.Close()
+			require.NoError(t, err)
+		})
+		b, err := ioutil.ReadAll(resp.Body)
+		require.NoError(t, err)
+		require.Equal(t, 200, resp.StatusCode)
+
+		require.JSONEq(t, expectedNoneJSON, string(b))
+	}
+
+	// Now, let's check we get the same rule when filtering by dashboard_uid and panel_id
+	{
+		promRulesURL := fmt.Sprintf("http://grafana:password@%s/api/ruler/grafana/api/v1/rules?dashboard_uid=%s&panel_id=1", grafanaListedAddr, dashboardUID)
+		// nolint:gosec
+		resp, err := http.Get(promRulesURL)
+		require.NoError(t, err)
+		t.Cleanup(func() {
+			err := resp.Body.Close()
+			require.NoError(t, err)
+		})
+		b, err := ioutil.ReadAll(resp.Body)
+		require.NoError(t, err)
+		require.Equal(t, 200, resp.StatusCode)
+
+		body, _ := rulesNamespaceWithoutVariableValues(t, b)
+		require.JSONEq(t, expectedFilteredByJSON, body)
+	}
+
+	// Now, let's check we get no rules when filtering by dashboard_uid and unknown panel_id
+	{
+		promRulesURL := fmt.Sprintf("http://grafana:password@%s/api/ruler/grafana/api/v1/rules?dashboard_uid=%s&panel_id=2", grafanaListedAddr, dashboardUID)
+		// nolint:gosec
+		resp, err := http.Get(promRulesURL)
+		require.NoError(t, err)
+		t.Cleanup(func() {
+			err := resp.Body.Close()
+			require.NoError(t, err)
+		})
+		b, err := ioutil.ReadAll(resp.Body)
+		require.NoError(t, err)
+		require.Equal(t, 200, resp.StatusCode)
+
+		require.JSONEq(t, expectedNoneJSON, string(b))
+	}
+
+	// Now, let's check we get that panel_id without dashboard_uid does not filter rules
+	{
+		promRulesURL := fmt.Sprintf("http://grafana:password@%s/api/ruler/grafana/api/v1/rules?panel_id=2", grafanaListedAddr)
+		// nolint:gosec
+		resp, err := http.Get(promRulesURL)
+		require.NoError(t, err)
+		t.Cleanup(func() {
+			err := resp.Body.Close()
+			require.NoError(t, err)
+		})
+		b, err := ioutil.ReadAll(resp.Body)
+		require.NoError(t, err)
+		require.Equal(t, 200, resp.StatusCode)
+
+		body, _ := rulesNamespaceWithoutVariableValues(t, b)
+		require.JSONEq(t, expectedAllJSON, body)
+	}
+
+	// Now, let's check an invalid panel_id returns a 400 Bad Request response
+	{
+		promRulesURL := fmt.Sprintf("http://grafana:password@%s/api/ruler/grafana/api/v1/rules?dashboard_uid=%s&panel_id=invalid", grafanaListedAddr, dashboardUID)
+		// nolint:gosec
+		resp, err := http.Get(promRulesURL)
+		require.NoError(t, err)
+		t.Cleanup(func() {
+			err := resp.Body.Close()
+			require.NoError(t, err)
+		})
+		require.Equal(t, http.StatusBadRequest, resp.StatusCode)
```

Comment:
CAn we check the body here? I'd love to understand how these errors look like and wether they're helpful or not.